### PR TITLE
Add warning when disabling secure.enable_security

### PR DESCRIPTION
**Pull request by rubenwardy**
Disabling mod security is never needed as mods can use `minetest.request_insecure_environment()` instead.

Showing a warning is the bare minimum, removing the setting should be considered at a later date

### CODE REVIEW

**Comment by appgurueu** [comment]
> Disabling mod security is never needed as mods can use minetest.request_insecure_environment() instead.

1. There are older mods which probably don't make use of this and won't be changed for various reasons (compatibility)
2. Does the insecure environment allow using `require`? IIRC it does not.

**Comment by rubenwardy** [comment]
1. this is a deprecation, so that doesn't matter. What mods are there?
2. Yes, it does

File: builtin/settingtypes.txt [hunk 1]
```diff
@@ -1323,7 +1323,7 @@ csm_restriction_noderange (Client side node lookup range restriction) int 0
 
 [*Security]
 
-#    Prevent mods from doing insecure things like running shell commands.
+#    Prevent mods from doing insecure things like running shell commands. Deprecated: Disabling this should never be done.
```
Comment (SmallJoker) [inline builtin/settingtypes.txt:1326]: Line too long

File: src/script/scripting_server.cpp [hunk 2]
```diff
@@ -62,6 +62,9 @@ ServerScripting::ServerScripting(Server* server):
 
 	if (g_settings->getBool("secure.enable_security")) {
 		initializeSecurity();
+	} else {
+		warningstream << "Disabling mod security is deprecated, and should not be done. "
+				<< "Mods should use minetest.request_insecure_environment() instead" << std::endl;
```
Comment (SmallJoker) [inline src/script/scripting_server.cpp:67]:
```suggestion
		warningstream << "Disabling mod security is deprecated and "
			<< "will be removed in a future release. Mods should use "
			<< "minetest.request_insecure_environment() instead" << std::endl;
```

Comment (rubenwardy) [reply src/script/scripting_server.cpp:67]: sfan5 is against removing this setting

How about:

> \!/ Disabling mod security should not be done, as it allows any mod to access the host computer. Mods should use minetest.request_insecure_environment() instead \!/

Comment (SmallJoker) [reply src/script/scripting_server.cpp]: that's okay as well.

**Comment by nerzhul** [comment]
i agree for setting depreciation too. Older mods which are not compatible should be considered abandonned and may be forked/fixed. We have mod security since very long time ago now.
I suggest that the warning talk about the depreciation too

**Comment by sfan5** [comment]
> Disabling mod security is never needed as mods can use `minetest.request_insecure_environment()` instead.

I don't disapprove of this PR, but here's a counterpoint:
Minetest-specific APIs such as `minetest.mkdir` or `minetest.safe_file_write` can not be made to work on arbitrary paths even if you have an insecure environment.

File: src/script/scripting_server.cpp [hunk 3]
```diff
@@ -62,6 +62,10 @@ ServerScripting::ServerScripting(Server* server):
 
 	if (g_settings->getBool("secure.enable_security")) {
 		initializeSecurity();
+	} else {
+		warningstream << "\\!/ Disabling mod security should not be done, as it allows any mod to "
```
Comment (rubenwardy) [inline src/script/scripting_server.cpp:66]:
```suggestion
		warningstream << "\\!/ Mod security should never be disabled, as it allows any mod to "
```

File: src/script/scripting_server.cpp [hunk 4]
```diff
@@ -62,6 +62,10 @@ ServerScripting::ServerScripting(Server* server):
...
+				<< "access the host computer."
```
Comment (rubenwardy) [inline src/script/scripting_server.cpp:67]:
```suggestion
				<< "access the host machine."
```

Comment (sfan5) [reply src/script/scripting_server.cpp:67]:
```suggestion
				<< "access the host machine!"
```

File: builtin/settingtypes.txt [hunk 5]
```diff
@@ -1324,6 +1324,7 @@ csm_restriction_noderange (Client side node lookup range restriction) int 0
 [*Security]
 
 #    Prevent mods from doing insecure things like running shell commands.
+#    Deprecated: Disabling this should never be done.
```
Comment (rubenwardy) [inline builtin/settingtypes.txt:1327]:
```suggestion
```

**Pull request merged**